templates/al2023/provisioners/install-nvidia-driver.sh

```diff
@@ -24,6 +24,13 @@ function rpm_install() {
 
 echo "Installing NVIDIA ${NVIDIA_DRIVER_MAJOR_VERSION} drivers..."
 
+# The AL2023 GPU AMI currently builds and archives the following nvidia kernel modules
+# in /var/lib/dkms-archive: nvidia, nvidia-open, nvidia-open-grid. To maintain the stability
+# of the AMI, we want to ensure that all three kernel modules (and also the userspace modules)
+# are on the same NVIDIA driver version. Currently, the script installs the NVIDIA GRID drivers
+# first and decides the full NVIDIA driver version that the AMI will adhere to
+NVIDIA_DRIVER_FULL_VERSION=""
```

Comment:
I think we should set the version here and then re-use it in forming the GRID path. This makes it a bit clearer how the version was determined, and it's also necessary for the arm64 build since there's a short circuit on the grid driver install

I don't expect that two minor versions of the same driver would be maintained at the same time, but to be safe, it might be better to do a strict version comparison rather than incorporating the time, e.g. :
```suggestion
NVIDIA_DRIVER_FULL_VERSION=$(aws s3 ls --recursive s3://ec2-linux-nvidia-drivers/ \
  | grep -Eo "(NVIDIA-Linux-x86_64-)$NVIDIA_DRIVER_MAJOR_VERSION\.[0-9]+\.[0-9]+(-grid-aws\.run)" \
  | cut -d'-' -f4 \
  | sort -V \
  | tail -1)
```